charts/m4d/templates/m4d-config.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: m4d-config
+data:
+  SECRET_PROVIDER_URL: "http://secret-provider.m4d-system:5555/get-secret"
+  SECRET_PROVIDER_ROLE: "demo"
+
+  {{- with .Values.vault }}
+  VAULT_ADDRESS: {{ .internal.vault_address }}
+  VAULT_DATASET_HOME: {{ .internal.dataset_home }}
+  VAULT_USER_HOME: {{ .internal.user_home }}
+  VAULT_DATASET_MOUNT: {{ .internal.dataset_mount }}
+  VAULT_USER_MOUNT: {{ .internal.user_mount }}
+  VAULT_TTL: {{ .internal.ttl }}
+  VAULT_AUTH: "kubernetes"
+  USER_VAULT_ADDRESS: {{ .external.user_vault_address }}
+  USER_VAULT_PATH: {{ .external.user_vault_path }}
```

Comment:
Do we still need the "external" parts?

Reply:
What are the plans for adding the rest of the config, e.g. catalog connector service, policy manager options?

Reply:
@shlomitk1 I am not sure about external. I can revisit this once the dateset credential work will progress if that makes sense. The rest of the vars in the config should be added as well as mentioned it's a starting point for writing the config map...

Reply:
I would remove internal/external. I was aiming for something like `config.vault.address`. Which of the parameters in values.yaml will survive after you finish the credentials work?

Reply:
We also need the rest of the parameters like Shlomit mentioned. The configmap is unusable without them. You can keep them the same as in config/prod for this PR.

Reply:
OK sure will add them.

Reply:
Regarding Vault vars: I left the VAULT_DATASET_HOME and VAULT_DATASET_MOUNT as is as I would like to keep the current implementation in the manager that enables the secret engine in VAULT_DATASET_MOUNT path and adds policy to read the secrets. This policy will be used when configuring each cluster by adding a role for the modules to use this policy.
user credentials (VAULT_USER_HOME VAULT_USER_MOUNT) will be store as secrets so I removed them. 
I also removed VAULT_AUTH as its cluster specific.
I am not sure about USER_VAULT_PATH and USER_VAULT_ADDRESS - it is to external vault instance-- do we want to keep that?

Reply:
Keep VAULT_USER_HOME and VAULT_USER_MOUNT for now, as manager will store the secret in the vault (wkc connector does not work with secrets directly)

Reply:
If the patch is OK I suggest to rename OPA_SERVER_URL to MAIN_POLICY_MANAGER_URL and EGERIA_SERVER_URL to CATALOG_PROVIDER_URL in m4d code.